packages/astro/src/vite-plugin-astro-server/route.ts

```diff
@@ -95,6 +95,19 @@ export async function matchRoute(
 	}
 
 	const custom404 = getCustom404Route(manifestData);
+	
+	if (custom404 && custom404.component === 'astro-default-404') {
+		const component: any = {
```

Comment:
Can we avoid the `any` and use `@ts-expect-error`, so we can add a comment that explains why we're doing this?

Reply:
The types around these parts are wishy washy, so you'd see a lot of any's. In this case, the correct type error'd. I'll  take another look.